src/frontend/calendar/cell.js

```diff
@@ -31,6 +33,20 @@ function CalendarCell( { blank = false, year, month, day, events } ) {
 					</h3>
 				);
 			} ) }
+
+			{ dayEvents.length > maxEventsToDisplay && (
+				<a
+					href={ '/day/' + date.toISOString() }
+					onClick={ ( e ) => {
+						e.preventDefault();
+
+						// @todo trigger list view
+					} }
+				>
+					{ dayEvents.length - maxEventsToDisplay }
+					{ __( 'more', 'wporg' ) }
```

Comment:
You can use `sprintf` & `_n` from the i18n package to make sure the string is translated with the number in the correct place & the correct tenses for different counts (not an issue in English, but I'm sure it's different in other languages). Like so:
```js
const count = dayEvents.length - maxEventsToDisplay;

{ sprintf( _n( '%d more', '%d more', count, 'text-domain' ),  count ) }
```